### ISSUE TITLE
dependency injection fails for \Magento\ConfigurableProduct\Pricing\Price\PriceResolverInterface

### ISSUE THREAD

**Issue by ioweb-gr** [magento/magento2#23717]
Unlike other interface classes the \Magento\ConfigurableProduct\Pricing\Price\PriceResolverInterface doesn't seem to be usable via dependency injection.

### Preconditions (*)
1. Magento 2.3.1

### Steps to reproduce (*)
1. Create a new module and a class
2. Add in the constructor the \Magento\ConfigurableProduct\Pricing\Price\PriceResolverInterface
e.g.
```

   public function __construct(
        ProductRepositoryInterface $productRepository,
        CategoryRepositoryInterface $categoryRepository,
        LoggerInterface $logger,
        SearchCriteriaBuilderFactory $searchCriteriaBuilderFactory,
        CollectionFactory $collectionFactory,
        Product $productModel,
        Data $helper,
        \Magento\Framework\Filesystem\DirectoryList $directoryList,
        StoreManagerInterface $storeManager,
        \Magento\ConfigurableProduct\Model\Product\Type\Configurable $configurableType,
        State $state,
        PriceResolverInterface $configurableRegularPrice
    )
```
### Expected result (*)
1. The dependency is injected correctly

### Actual result (*)
1. PHP Error 
PHP Fatal error:  Uncaught Error: Cannot instantiate interface Magento\ConfigurableProduct\Pricing\Price\PriceResolverInterface in /private/magento/2x/default/httpdocs/vendor/magento/framework/ObjectManager/Factory/Dynamic/Developer.php:50
Stack trace:
#0 /private/magento/2x/default/httpdocs/vendor/magento/framework/ObjectManager/ObjectManager.php(70): Magento\Framework\ObjectManager\Factory\Dynamic\Developer->create('Magento\\Configu...')
#1 /private/magento/2x/default/httpdocs/vendor/magento/framework/ObjectManager/Factory/AbstractFactory.php(160): Magento\Framework\ObjectManager\ObjectManager->get('Magento\\Configu...')
#2 /private/magento/2x/default/httpdocs/vendor/magento/framework/ObjectManager/Factory/AbstractFactory.php(246): Magento\Framework\ObjectManager\Factory\AbstractFactory->resolveArgument(Array, 'Magento\\Configu...', NULL, 'configurableReg...', 'Vendor\\Module...')
#3 /private/magento/2x/default/httpdocs/vendor/magento/framework/ObjectManager/Factory/Dynamic/Developer.php(34): Magento\Framework\ObjectManage in /private/magento/2x/default/httpdocs/vendor/magento/framework/ObjectManager/Factory/Dynamic/Developer.php on line 50

**Comment by arushi013** [comment]
Issue is reproducible in Magento 2.3-develop and is coming because no preference are added for PriceResolverInterface.

**Comment by magento-engcom-team** [comment]
@edenduong Thank you for verifying the issue.

Unfortunately, not enough information was provided to acknowledge ticket. Please consider adding the following:
- [x] Add `"Component: "` label(s) to this ticket based on verification result. If uncertain, you may follow the best guess
- [x] Add `"Reproduced on "` label(s) to this ticket based on verification result

Once all required information is added, please add label `"Issue: Confirmed"` again.
Thanks!

**Comment by edenduong** [comment]
Yes, the reason because no preference of PriceResolverInterface in di.xm

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @shikhamis11 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-18893` were created

**Issue Available**: @shikhamis11, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by shikhamis11** [comment]
since the issue is fixed and merged to 2.3-develop branch I am closing this one
